Sources/Core/StringExtensions.swift

```diff
@@ -137,35 +137,119 @@ let objectiveCReservedWords = Set<String>([
     "while",
 ])
 
+// TODO: "id" is technically allowed. It's possible not everyone wants this replacement.
+let javaReservedWordReplacements = [
+    "id": "uid",
+]
+
+// https://en.wikipedia.org/wiki/List_of_Java_keywords
+let javaReservedWords = Set<String>([
+    "abstract",
+    "assert",
+    "boolean",
+    "break",
+    "byte",
+    "case",
+    "catch",
+    "char",
+    "class",
+    "continue",
+    "default",
+    "do",
+    "double",
+    "else",
+    "enum",
+    "exports",
+    "extends",
+    "final",
+    "finally",
+    "float",
+    "for",
+    "if",
+    "implements",
+    "import",
+    "instanceof",
+    "int",
+    "interface",
+    "long",
+    "module",
+    "native",
+    "new",
+    "package",
+    "private",
+    "protected",
+    "public",
+    "requires",
+    "return",
+    "short",
+    "static",
+    "strictfp",
+    "super",
+    "switch",
+    "synchronized",
+    "this",
+    "throw",
+    "throws",
+    "transient",
+    "try",
+    "void",
+    "volatile",
+    "while",
+    "true",
+    "null",
+    "false",
+    "var",
+    "const",
+    "goto",
+])
+
 extension String {
     func indent() -> String {
         // We indent with tabs and in a post process the tabs are changed to a specific number of spaces
         return "\t" + self
     }
 
     /// All components separated by _ will be capitalized including the first one
-    func snakeCaseToCamelCase() -> String {
+    func snakeCaseToCamelCase(_ language: Languages = Languages.objectiveC) -> String {
```

Comment:
Some idea: What if we would create a "Language" Enum / Struct that knows that language `Self` is, e.g. either Objective-C, Java or some other language and that would contain the logic around reserved word replacement and capabilities such as creating `snakeCaseToCamelCase`, `snakeCaseToPropertyName`, `snakeCaseToPropertyName`. We could have one instance of language and passing this around and so we don't have to have this if else branching in a String extension as well as you don't have to pass in `.java`, `.objectiveC` everywhere as parameter. You would just call `snakeCaseToCamelCase` on the Language object you have in the Renderer.

Reply:
Thanks. Moved to `Languages` enum.